test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/DacFx/DacFxserviceTests.cs

```diff
@@ -159,6 +160,40 @@ public async void ExtractDacpac()
             }
         }
 
+        /// <summary>
+        /// Verify the extract request to create Sql file
+        /// </summary>
+        [Fact]
+        public async void ExtractDBtoSqlFile()
+        {
+            var result = GetLiveAutoCompleteTestObjects();
+            SqlTestDb testdb = await SqlTestDb.CreateNewAsync(TestServerType.OnPrem, false, null, SourceScript, "DacFxExtracttoSqlTest");
+            string folderPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "DacFxTest");
+            Directory.CreateDirectory(folderPath);
+
+            try
+            {
+                var extractParams = new ExtractParams
+                {
+                    DatabaseName = testdb.DatabaseName,
+                    PackageFilePath = Path.Combine(folderPath, string.Format("{0}.sql", testdb.DatabaseName)),
+                    ApplicationName = "test",
+                    ApplicationVersion = "1.0.0.0",
+                    ExtractTarget = DacExtractTarget.File 
+                };
+
+                DacFxService service = new DacFxService();
+                ExtractOperation operation = new ExtractOperation(extractParams, result.ConnectionInfo);
+                service.PerformOperation(operation, TaskExecutionMode.Execute);
+
+                VerifyAndCleanup(extractParams.PackageFilePath);
```

Comment:
Does it know how to check that it actually extracted to a file rather than mistakenly to a dacpac that's been named .sql?  Maybe the test should extract to Flat (multiple files, right?) and validate that the correct number of *.sql were created as a higher confidence test.

Reply:
Added another test to extract to Flat target.